### PR TITLE
🐛  Fix EXC_BAD_ACCESS crash on device

### PR DESCRIPTION
**Pull request by fillito**
### :pushpin: References
* **Issue:** https://github.com/pedrovgs/DeepPaneliOS/issues/12

### :soccer: What is the goal?

- Fixes a crash that only occurs on physical devices.
- Allows disabling the usage of Metal 

### How is it being implemented?

I've removed the `UnsafeMutablePointer` in `mapOutputTensorToPredicition`  and passed a reference to a now mutable `prediction` array. This solves, what I think is, a premature deallocation of the `logits` array. 

Additionally, I've added a new `useMetal: Bool` parameter that allows us to disable Metal. 

### How can it be tested?

:robot:
Manually tested analyzing the images from the Sample App.

### CODE REVIEW

File: DeepPanel.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -682,6 +683,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
```
Comment (fillito) [inline DeepPanel.xcodeproj/project.pbxproj:686]: ℹ️   This was changed automatically by Xcode 12 to silence some new warnings.

File: DeepPanel/DeepPanel.swift [hunk 2]
```diff
@@ -82,7 +82,7 @@ public class DeepPanel {
               options?.threadCount = 2
         #else
               // Use GPU on real device for inference as this model is fully supported.
-              delegates = [MetalDelegate()]
+              delegates = useMetal ? [MetalDelegate()] : nil
```
Comment (juanjo-ramos) [inline DeepPanel/DeepPanel.swift:85]: Wouldn't you want to
> // Use CPU for inference as MetalDelegate does not support iOS simulator.

in case Metal is not supported? As in, I would expect that you need to use something for inference and this option is using `nil` if Metal isn't supported on a real device.

**Review by juanjo-ramos — COMMENTED** [review]
LGTM. I just have a question regarding what inference to use when running on a real device that doesn't support Metal.

**Review by pedrovgs — APPROVED** [review]

**Comment by pedrovgs** [comment]
Thanks @fillito, I really appreciate you took time to review and fix this issue. I'll release a new version today 😃

File: DeepPanel/DeepPanel.swift (hunk 2 above)
Comment (fillito) [reply DeepPanel/DeepPanel.swift:85]: As far as I know, Metal is a replacement for OpenGL and does not require any particular hardware. Any device running iOS 8 and higher can run Metal so, unless I'm mistaken, we won't find any device not supporting Metal. [Reference](https://developer.apple.com/library/archive/documentation/DeviceInformation/Reference/iOSDeviceCompatibility/HardwareGPUInformation/HardwareGPUInformation.html#//apple_ref/doc/uid/TP40013599-CH106-SW1)

This flag is just a workaround until we understand better why sometimes we don't get any results when using Metal. 

It's true that, when the flag is `true`, the comment above is not accurate. But I think we'll be able to completely get rid of the flag once we know the details.
